File: tests/fixtures/sample_code.py
import sys


def retry_request(url, max_attempts=3):
    """Retry a request up to max_attempts times"""
    for attempt in range(max_attempts):
        try:
            return fetch(url)
        except RequestError:
            if attempt == max_attempts - 1:
                raise
    return None


class RetryHandler:
    def __init__(self, max_retries=5):
        self.max_retries = max_retries

    def handle_retry(self, operation):
        """Handle retry logic for an operation"""
        for i in range(self.max_retries):
            try:
                return operation()
            except Exception as e:
                if i == self.max_retries - 1:
                    raise
                print(f"Retry attempt {i + 1} failed: {e}")

File: tests/fixtures/sample_code.js
function retryFetch(url, maxAttempts = 3) {
    for (let i = 0; i < maxAttempts; i++) {
        try {
            return fetch(url);
        } catch (error) {
            if (i === maxAttempts - 1) throw error;
        }
    }
}

class RetryManager {
    constructor(maxRetries = 5) {
        this.maxRetries = maxRetries;
    }

    async executeWithRetry(operation) {
        for (let attempt = 0; attempt < this.maxRetries; attempt++) {
            try {
                return await operation();
            } catch (error) {
                if (attempt === this.maxRetries - 1) throw error;
                console.log(`Retry attempt ${attempt + 1} failed`);
            }
        }
    }
}
